src/frontend/react-hiringsystem/src/components/shared/JobContext.js

```diff
@@ -0,0 +1,34 @@
+import axios from "axios";
+import { createContext, useState } from "react";
+import { useNavigate } from "react-router-dom";
+
+const JobContext = createContext();
+
+export const CreateJobContextProvider = ({ children }) => {
```

Comment:
Wouldn't JobContextProvider be a more suitable name, as its purpose is not creational only?

Reply:
Done. As mentioned also in the above answer, this name mistakenly remained from the initial version, with separate contexts.